packages/protocol/contracts/governance/Election.sol

```diff
@@ -729,50 +798,30 @@ contract Election is IElection, Ownable, ReentrancyGuard, Initializable, UsingRe
   function getTotalVotesForEligibleValidatorGroups()
     external
     view
-    returns (address[] memory, uint256[] memory)
+    returns (address[] memory groups, uint256[] memory values)
   {
     return votes.total.eligible.getElements();
   }
 
-  function validatorAddressFromCurrentSet(uint256 index) external view returns (address) {
-    address validatorAddress;
-    assembly {
-      let newCallDataPosition := mload(0x40)
-      mstore(newCallDataPosition, index)
-      let success := staticcall(5000, 0xfa, newCallDataPosition, 32, 0, 0)
-      returndatacopy(add(newCallDataPosition, 64), 0, 32)
-      validatorAddress := mload(add(newCallDataPosition, 64))
-    }
-
-    return validatorAddress;
-  }
-
-  function numberValidatorsInCurrentSet() external view returns (uint256) {
-    uint256 numberValidators;
-    assembly {
-      let success := staticcall(5000, 0xf9, 0, 0, 0, 0)
-      let returnData := mload(0x40)
-      returndatacopy(returnData, 0, 32)
-      numberValidators := mload(returnData)
-    }
-
-    return numberValidators;
-  }
-
   /**
    * @notice Returns a list of elected validators with seats allocated to groups via the D'Hondt
    *   method.
    * @return The list of elected validators.
    * @dev See https://en.wikipedia.org/wiki/D%27Hondt_method#Allocation for more information.
    */
   function electValidators() external view returns (address[] memory) {
-    // Only members of these validator groups are eligible for election.
-    uint256 maxNumElectionGroups = Math.min(
-      electableValidators.max,
-      votes.total.eligible.list.numElements
+    // Groups must have at least `electabilityThreshold` proportion of the total votes to be
+    // considered for the election.
+    uint256 requiredVotes = electabilityThreshold.multiply(
+      FixidityLib.newFixed(getTotalVotes())
+    ).fromFixed();
+    // Only consider groups with at least `requiredVotes` but do not consider more groups than the
+    // max number of electable validators.
+    uint256 numElectionGroups = votes.total.eligible.numElementsGreaterThan(
```

Comment:
seems like you have a `2*N` and can be done in `N`. For example, something like a `takeUntil()` get all electionGroups that meet the requiredVotes criteria